submariner-operator/submariner-metrics-redesign.md

```diff
@@ -0,0 +1,163 @@
+# Submariner metrics redesign
+
+Related Issue:
+[Exporting Submariner metrics without needing to open ports](https://github.com/submariner-io/submariner/issues/1944)
+
+## Summary
+
+Submariner Gateway and Globanet pods run with HostNetworking. To access metrics exported by these two we need to open ports
+in the infra firewall. Current design has a limitation where these ports might be in use by some other service,
+causing a conflict and metrics becoming inaccessible. With the move towards minimizing changes needed in cloud infra for
+submariner, there is a use case to export metrics in a way that doesn't require any of these ports to be opened.
+
+## Proposal
+
+This enhancement proposes a redesign of existing metrics for Submariner Gateway and Globalnet by adding a new metrics proxy
+service. Pods running on the same node can access ports without requiring ports to be opened in the cloud infra. Any queries
+for these metrics will land on the proxy which will always run on the same node.
+
+This proposal also recommends changing the port used by metrics http servers in gateway and globalnet pods to a free range
+port to avoid problem of port being already in use on host network.
+
+## Design Details
+
+* A new DaemonSet `submariner-metrics-proxy` will be created. It will use the NodeSelector `submariner.io/gateway=true`
+to run on all gateway nodes, active or not.
+* DaemonSet will also use toleration to make sure it can always be scheduled on gateway node, irrespective of any taints.
+* `submariner-metrics-proxy` Pod will use `netcat` to act as proxy and forward all incoming requests on metrics port to the
+metrics port on the node IP.
+* The proxy pod will run two containers if globlanet is enabled, one each for gateway and globalnet.
+* Existing metrics service for gateway and globalnet will be modified to use the new metrics proxy pod. This will be
+done by using same label selector as metrics proxy DaemonSet. This will make sure users can continue using existing services
+and ports without needing changes.
```

Comment:
```suggestion
* The existing metrics service for Gateway and Globalnet will be modified to use the new metrics proxy pod. This will be
done by using same label selector as the metrics proxy DaemonSet. This will make sure users can continue using existing services
and ports without needing to make any changes themselves.
```